challenge-manager/chain-watcher/watcher.go

```diff
@@ -1056,3 +1060,7 @@ func (w *Watcher) saveEdgeToDB(
 		HasLengthOneRival:   hasLengthOneRival,
 	})
 }
+
+func daysToBlocks(avgBlockCreationTime time.Duration, days uint64) uint64 {
```

Comment:
Maybe this should just be `blocksPerInterval(avgBlockTime time.Duration, int time.Duration)` and it could be called like this: `blocksPerInterval(averageTimeForBlockCreation, time.Hour * 24 * 21)`
Or even pull `time.Hour * 24` into a package constant `Day`?

I think it would make the implementation prettier because you can divide one `time.Duration` by another and not have to deal with the floating point and rounding.